dts/bindings/sensor/ams,tmd26xx.yaml

```diff
@@ -0,0 +1,99 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 Carl Zeiss Meditec AG
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    OSRAM ams TMD26XX Proximity Sensor
+
+compatible: "ams,tmd26xx"
+
+include: [sensor-device.yaml, i2c-device.yaml]
```

Comment:
you could probably also include ams, tmd2620.yaml over here, and then you need to add only the new properties that are not already present in tmd2620, since tmd2620 properties are a subset of the tmd26xx properties.